### ISSUE TITLE
Incorrect Year assigned to New Application

### ISSUE THREAD

**Issue by dandahle** [dandahle/Catalyst-AppVetting#145]
In the Production Database Application A1959-1 shows an application date with the year 1959 (hence the Application #). How did this date and application # get assigned by the application?

**Comment by rarafon** [comment]
I also got this on my local server

![image](https://user-images.githubusercontent.com/11997170/83815411-faa8b900-a674-11ea-8726-9f4574f14f51.png)

**Comment by rarafon** [comment]
Found out why I think.

Application year is based off of the year from the signature date:
![image](https://user-images.githubusercontent.com/11997170/83815579-570bd880-a675-11ea-8697-73aed4af1009.png)

**Comment by rarafon** [comment]
Also, I don't know if it was intentional, but the "Application date" is actually the signature date. I originally thought it was the creation date of the application.

**Comment by dandahle** [comment]
Hi Andy - Thanks for checking on this and solving the puzzle! I always assumed that they used the system date stamp for creating the Application number, not the applicant’s date entry for signing. It makes more sense, then, that we would have an A1929-1 application like you have the A1010-1 application. 

There is also an issue #146 to make the application numbers 3 digits so they sort properly. Right now they sort by the first number they see so Ayyyy-10 comes before Ayyyy-9 when sorting by the application # column. 

Would you be willing to take on fixing those Issues after your term is complete? 

Thank you, 

==> Dan

**Comment by rarafon** [comment]
Hi Dan,
I linked a PR that I made #166 , which used the application's creation date instead of the signature date.

I also noticed that in all the packages/models, the created and updated property were all created and updated manually. Mongoose has the timestamps options which does this automatically for you. 

Here's a stack overflow q&a about it: https://stackoverflow.com/questions/10006218/which-schematype-in-mongoose-is-best-for-timestamp

The answer by bobbyz (2nd answer) explains it in detail. It was what I used for the Care Network.

I'll also take look at issue #146 .
